src/LinuxTracing/LinuxTracingUtils.cpp

```diff
@@ -58,14 +47,19 @@ std::optional<char> GetThreadState(pid_t tid) {
     return std::nullopt;
   }
 
+  ErrorMessageOr<std::string> file_content = orbit_base::ReadFileToString(stat);
   std::ifstream stream{stat.string()};
-  if (!stream.good()) {
-    ERROR("Could not open \"%s\"", stat.string());
+  if (!file_content) {
+    ERROR("Could not open \"%s\": %s", stat.string(), file_content.error().message());
     return std::nullopt;
   }
 
-  std::string first_line{};
-  std::getline(stream, first_line);
+  std::vector<std::string> lines = absl::StrSplit(file_content.value(), absl::MaxSplits('\n', 1));
+  if (lines.empty()) {
+    ERROR("Empty \"%s\" file.", stat.string());
```

Comment:
```suggestion
    ERROR("Empty \"%s\" file", stat.string());
```
(we usually don't use periods in logs)